src/auth/auth.module.ts

```diff
@@ -0,0 +1,27 @@
+import { Module } from '@nestjs/common';
+import { JwtModule, JwtService } from '@nestjs/jwt';
+import { PassportModule } from '@nestjs/passport';
+import { AuthService } from './auth.service';
+import { AuthController } from './auth.controller';
+import { RedisModule } from '@liaoliaots/nestjs-redis';
+import { Env } from 'src/env';
+import { JwtStrategy } from './jwt.strategy';
+import { GithubModule } from 'src/github/github.module';
+
+@Module({
+  imports: [
+    GithubModule,
+    PassportModule,
+    JwtModule.register({
+      secret: Env.JWT.AUTH_TOKEN,
+      signOptions: { expiresIn: Env.JWT.EXPIRES_IN },
+    }),
+    RedisModule.forRoot({
+      config: { host: Env.REDIS.HOST, port: Env.REDIS.PORT },
+    }),
```

Comment:
PullProofAI

The `JwtModule.register` configuration can be extracted into a separate method for better readability and maintainability.
```diff
+     JwtModule.registerAsync({
+       useFactory: () => ({
+         secret: Env.JWT.AUTH_TOKEN,
+         signOptions: { expiresIn: Env.JWT.EXPIRES_IN },
+       }),
+     }),
```

23:
Consider using an interface to define the shape of providers instead of directly listing them in the `providers` array.
```diff
-   providers: [AuthService, JwtStrategy],
+   providers: [AuthService, JwtStrategy] as const,
```

25:
It's good practice to explicitly specify the return type of the `exports` property for clarity.
```diff
-   exports: [AuthService],
+   exports: [AuthService] as const,
```

### End of Review


<!-- This is an auto-generated comment by PullProofAI -->